product_code_unique/product.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+
+from openerp.osv import osv
+from openerp.tools.translate import _
+
+
+class product_code_unique_product(osv.osv):
+    _name = "product.product"
+    _inherit = "product.product"
+
+    def _check_defaultcode_and_variants(self, cr, uid, ids):
+        for product in self.browse(cr, uid, ids):
```

Comment:
@clonedagain You are right, it's the official answer from Odoo https://www.odoo.com/fr_FR/forum/help-1/question/solved-browse-vs-read-performance-67752. I am, however, quite skeptical and I often wonder if Odoo developers sometimes look at the number of SQL queries generated by the orm.